src/Workspaces/Core/Portable/Workspace/Solution/Checksum.cs

```diff
@@ -20,16 +20,18 @@ namespace Microsoft.CodeAnalysis
     /// Checksum of data can be used later to see whether two data are same or not
     /// without actually comparing data itself
     /// </summary>
-    [DataContract]
-    internal sealed partial record class Checksum(
-        [property: DataMember(Order = 0)] Checksum.HashData Hash)
+    [DataContract, StructLayout(LayoutKind.Explicit, Size = HashSize)]
+    internal readonly partial record struct Checksum(
```

Comment:
>Checksum [](http://example.com/codeflow?start=44&length=8)

Perhaps rename Checksum to ContentChecksum to avoid collisions with Checksum properties?